csum-file.c

```diff
@@ -146,9 +157,35 @@ struct hashfile *hashfd_throughput(int fd, const char *name, struct progress *tp
 	f->name = name;
 	f->do_crc = 0;
 	the_hash_algo->init_fn(&f->ctx);
+
+	f->buffer_len = buffer_len;
+	f->buffer = xmalloc(buffer_len);
+	f->check_buffer = NULL;
```

Comment:
Should we `FREE_AND_NULL(f->check_buffer)` here?

Reply:
We are initializing the struct here, not freeing it.

Reply:
Oh right!